common/ledger/blkstorage/blockfile_mgr.go

```diff
@@ -757,49 +761,49 @@ type blockfilesInfo struct {
 }
 
 func (i *blockfilesInfo) marshal() ([]byte, error) {
-	buffer := proto.NewBuffer([]byte{})
-	var err error
-	if err = buffer.EncodeVarint(uint64(i.latestFileNumber)); err != nil {
-		return nil, errors.Wrapf(err, "error encoding the latestFileNumber [%d]", i.latestFileNumber)
-	}
-	if err = buffer.EncodeVarint(uint64(i.latestFileSize)); err != nil {
-		return nil, errors.Wrapf(err, "error encoding the latestFileSize [%d]", i.latestFileSize)
-	}
-	if err = buffer.EncodeVarint(i.lastPersistedBlock); err != nil {
-		return nil, errors.Wrapf(err, "error encoding the lastPersistedBlock [%d]", i.lastPersistedBlock)
-	}
+	var buf []byte
+	buf = protowire.AppendVarint(buf, uint64(i.latestFileNumber))
+	buf = protowire.AppendVarint(buf, uint64(i.latestFileSize))
+	buf = protowire.AppendVarint(buf, i.lastPersistedBlock)
 	var noBlockFilesMarker uint64
 	if i.noBlockFiles {
 		noBlockFilesMarker = 1
 	}
-	if err = buffer.EncodeVarint(noBlockFilesMarker); err != nil {
-		return nil, errors.Wrapf(err, "error encoding noBlockFiles [%d]", noBlockFilesMarker)
-	}
-	return buffer.Bytes(), nil
+	buf = protowire.AppendVarint(buf, noBlockFilesMarker)
+
+	return buf, nil
```

Comment:
itit will always return the nil error value, IMO you'd better remove it.

Reply:
fix